doc/release/upcoming_changes/25789.c_api.rst

```diff
@@ -0,0 +1,3 @@
+* The C-API now only exports the static inline function versions
+  of the array accessors (previously this dependent on using "deprecated API").
+  While we encourage avoiding it, if you need to use the struct fields directly.
```

Comment:
If committing this from the github UI, don't forget to add [skip cirrus] [skip azp] [skip actions] to the commit message

```suggestion
  While we encourage avoiding it, you can still use the struct fields directly.
```